Release notes and bump version for v0.31.1 (#1910)
loadimpact/k6: release notes/v0.31.1.md

> k6 v0.31.1 is a minor release with a single bugfix.

Reviewer: ```suggestion
k6 v0.31.1 is a patch release with a single bugfix.
```

> The bugfix is about the cloud output and the new `http_req_failed` metric. Due to additional state being used for the transport of it to the cloud output specifically, and a misunderstanding of what a functional call from a library dependency does, the `http_req_failed` was always set to `1`. This does not effect any other output or the end of summary as those are done through the standard metric emission logic and are also a lot easier to test and notice if wrong. ([#1908](https://github.com/loadimpact/k6/issues/1908))

Reviewer 2: ```suggestion
The bugfix is about the cloud output and the new `http_req_failed` metric. Due to additional state being used for the transport of it to the cloud output specifically, and a misunderstanding of what a functional call from a library dependency does, the `http_req_failed` was always set to `1`. This does not affect any other output or the end of summary as those are done through the standard metric emission logic and are also a lot easier to test and notice if wrong. ([#1908](https://github.com/loadimpact/k6/issues/1908))
```

> The bugfix is about the cloud output and the new `http_req_failed` metric. Due to additional state being used for the transport of it to the cloud output specifically, and a misunderstanding of what a functional call from a library dependency does, the `http_req_failed` was always set to `1`. This does not effect any other output or the end of summary as those are done through the standard metric emission logic and are also a lot easier to test and notice if wrong. ([#1908](https://github.com/loadimpact/k6/issues/1908))

Reviewer: ```suggestion
The bugfix is about the cloud output and the new `http_req_failed` metric in k6 v0.31.0. Due to additional state being used for its transport to the k6 cloud, and a misunderstanding of what a functional call from a library dependency does, the `http_req_failed` values were always set to `1`. This did not effect any other output or the end of test summary. ([#1908](https://github.com/loadimpact/k6/issues/1908))
```
